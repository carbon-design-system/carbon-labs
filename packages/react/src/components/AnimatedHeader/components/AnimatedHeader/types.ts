/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import type { BaseTileProps } from '../../components/Tiles/BaseTile/BaseTile';

export type Tile = BaseTileProps;

export interface TileGroup {
  id: number;
  label: string;
  tiles: Tile[];
}

export interface AriaLabels {
  welcome?: string;
  description?: string;
  collapseButton?: string;
  expandButton?: string;
  tilesContainer?: string;
}
