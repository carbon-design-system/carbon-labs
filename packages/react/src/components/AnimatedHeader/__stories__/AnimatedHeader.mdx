import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as AnimatedHeaderStories from './AnimatedHeader.stories';
import AnimatedHeader from '../components/AnimatedHeader/AnimatedHeader';

<Meta isTemplate />

# Animated Header

- **Initiative owner(s):** Drew Glapa
- **Status:** Draft
- **Target library:** TBD
- **Target library maintainer(s) / PR Reviewer(s):** N/A
- **Support channel:** `#carbon-labs`

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

> 💡 Check our
> [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/AnimatedHeader)
> example implementation.

[![Edit carbon-labs](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/AnimatedHeader)

## Table of Contents

- [Overview](#overview)
- [Getting started](#getting-started)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

The animated header is meant to be used on homepages only. It ensures
consistency throughout landing experiences, providing a cohesive appearance and
clear call to action for users. The skeleton for this header component ensures
consistency while flexible page elements allow teams to tailor the leadspace to
their individual user and product needs.

Lead spaces include a welcome text, a short description communicating the
products value proposition and feature most relevant content for the user.
Whether these are links to specific tasks, workspaces or other information is
defined by each product and their user’s needs. In product-imagery has to be
very settled and subtil to not distract users in performing their tasks.

<Canvas
  of={AnimatedHeaderStories.ThemeG10}
  additionalActions={[
    {
      title: 'Open in Stackblitz',
      onClick: () => stackblitzPrefillConfig(AnimatedHeaderStories.ThemeG10),
    },
  ]}
/>

### Anatomy

**Title:** Welcomes the user (Welcome, username)

**Content:** Short sentence in max. 3 lines related to product context; Action
button (optional), Expose content that need highest attention from the user or
content that trigger an action and allow users to directly start working and
gain value (within one click). It is recommended to display content on cards to
achieve highest contrast to the background.

**Visual / brand image:** Line-style illustration

**Collapse / Expand:** Icon to collapse or expand the header

### Theming

The default theme for Animated Header is _g10_ (Light theme).

Animated header supports both _g10_ (Light theme) and _g100_ (Dark Theme).

To use dark theme in your product/app wrap the componenet in the Carbon
[GlobalTheme](https://react.carbondesignsystem.com/?path=/docs/components-theme--overview#setting-the-global-theme)
component. Following the example from the Carbon website it would like this:

```
import { useEffect } from 'react';
import { GlobalTheme } from '@carbon/react';

function App() {
  const theme = 'g100'; // ← your implementation, e.g. fetching user settings

  useEffect(() => {
    document.documentElement.dataset.carbonTheme = theme;
  }, [theme]);

  return (
    <GlobalTheme theme={theme}>
      <AnimatedHeader />
    </GlobalTheme>;
  );
}
```

## Getting started

Here's a quick example to get you started.

```bash
yarn add @carbon/react
yarn add @carbon-labs/react-animated-header
```

### JS (via import)

```javascript
import { AnimatedHeader } from '@carbon-labs/react-animated-header/es/index';

function App() {
  return <AnimatedHeader />;
}

export default App;
```

### SCSS

In your styles file import

```
@use '@carbon/react';
@use '@carbon-labs/react-animated-header/scss/animated-header';
```

## Component API

<ArgTypes />

## Example usages

Example of Animated Header with all prop types enabled.

```jsx
  <AnimatedHeader
    tasksControllerConfig={tasksControllerConfigDropdown as TasksControllerConfig}
    workspaceSelectorConfig={WorkspaceSelectorConfig}
    description="Connect, monitor, and manage data."
    productName="[Product name]"
    userName="Drew"
    welcomeText="Welcome"
    isLoading={false}
    expandButtonLabel="Expand"
    collapseButtonLabel="Collapse"
    tileClickHandler={handleTileClicks}
  />
```
