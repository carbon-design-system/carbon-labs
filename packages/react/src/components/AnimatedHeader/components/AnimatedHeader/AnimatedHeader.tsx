/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import PropTypes from 'prop-types';
import React, {
  ElementType,
  ReactNode,
  useEffect,
  useRef,
  useState,
} from 'react';
import { Grid, Column, Button, Tooltip } from '@carbon/react';
import { ChevronUp, ChevronDown } from '@carbon/icons-react';
import lottie from 'lottie-web';
import { usePrefix } from '@carbon-labs/utilities/es/index.js';

import { BaseTile } from '../Tiles/index';
import TasksController, {
  TasksControllerProps,
} from '../TasksController/TasksController';
import WorkspaceSelector, {
  WorkspaceSelectorProps,
} from '../WorkspaceSelector/WorkspaceSelector';

/** Animated Header */

export interface Tile {
  href?: string | null;
  id: string;
  mainIcon?: ElementType | null;
  secondaryIcon?: ElementType | null;
  subtitle?: string | null;
  title?: string | null;
  customContent?: ReactNode | null;
  isLoading?: boolean;
  isDisabled?: boolean;
  isInteractive?: boolean;
}

export interface TileGroup {
  id: number;
  label: string;
  tiles: Tile[];
}

export type AnimatedHeaderProps = {
  allTileGroups?: TileGroup[];
  selectedTileGroup?: TileGroup;
  setSelectedTileGroup: (e) => void;
  description?: string;
  headerAnimation?: object;
  headerStatic?: React.JSX.Element;
  productName?: string;
  userName?: string;
  welcomeText?: string;
  isLoading?: boolean;
  disabledTaskLabel?: string;
  expandButtonLabel?: string;
  collapseButtonLabel?: string;
  tileClickHandler?: (tile: Tile) => void;
} & TasksControllerProps &
  WorkspaceSelectorProps;

const AnimatedHeader: React.FC<AnimatedHeaderProps> = ({
  allTileGroups,
  selectedTileGroup,
  setSelectedTileGroup,
  description,
  headerAnimation,
  headerStatic,
  productName = '[Product name]',
  userName,
  welcomeText,
  tasksControllerConfig,
  workspaceSelectorConfig,
  isLoading,
  disabledTaskLabel,
  expandButtonLabel = 'Expand',
  collapseButtonLabel = 'Collapse',
  tileClickHandler,
}: AnimatedHeaderProps) => {
  const prefix = usePrefix();
  const blockClass = `${prefix}--animated-header`;

  const animationContainer = useRef<HTMLDivElement>(null);
  const [open, setOpen] = useState(true);
  const [headingTextAnimation, setHeadingTextAnimation] = useState('');
  const isReduced = window.matchMedia('(prefers-reduced-motion)').matches;

  const collapsed = `${blockClass}--collapsed`;
  const contentCollapsed = `${blockClass}__content--collapsed`;
  const descriptionCollapsed = `${blockClass}__left-area-container--collapsed`;
  const headingCollapsed = `${blockClass}__heading--collapsed`;
  const headingExpanded = `${blockClass}__heading--expanded`;
  const lottieCollapsed = `${blockClass}__lottie-animation--collapsed`;

  const handleButtonCollapseClick = () => {
    setOpen(!open);

    open
      ? setHeadingTextAnimation(headingCollapsed)
      : setHeadingTextAnimation(headingExpanded);
  };

  useEffect(() => {
    if (!animationContainer.current) {
      return;
    }

    const animation = lottie.loadAnimation({
      container: animationContainer.current as HTMLDivElement,
      animationData: headerAnimation,
      renderer: 'svg',
      loop: false,
      autoplay: false,
    });

    const lottieLoadSpeed = 1;
    const animationData = animation['animationData'];

    function reducedMotion() {
      animation.goToAndStop(5000);
    }

    function load() {
      animation.setSpeed(lottieLoadSpeed);
      animation.playSegments(
        [animationData.markers.at(0).tm, animationData.markers.at(1).tm],
        true
      );
    }

    if (isReduced) {
      // reduced motion on
      animation.addEventListener('DOMLoaded', reducedMotion);
    } else {
      // reduced motion off

      // Run Start Animation
      animation.addEventListener('DOMLoaded', load);
    }
    return () => {
      animation?.removeEventListener('DOMLoaded', reducedMotion);
      animation?.removeEventListener('DOMLoaded', load);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isReduced]);

  return (
    <section className={`${blockClass}${!open ? ` ${collapsed}` : ''}`}>
      <Grid>
        <div className={`${blockClass}__gradient--overlay`} />

        <div className={`${blockClass}__container--gradient`} />

        {!headerAnimation && headerStatic && (
          <div className={`${blockClass}__static--container`}>
            <div
              className={`${blockClass}__static`}
              // eslint-disable-next-line react/forbid-dom-props
              style={{ backgroundImage: `url(${headerStatic})` }}
            />
          </div>
        )}

        <div className={`${blockClass}__lottie-animation--container`}>
          <div
            ref={animationContainer}
            className={`${blockClass}__lottie-animation${
              !open ? ` ${lottieCollapsed}` : ''
            }`}></div>
        </div>

        <Column sm={4} md={8} lg={16}>
          <Tooltip align="bottom" label={`${welcomeText}, ${userName}`}>
            <h1 className={`${blockClass}__heading ${headingTextAnimation}`}>
              <span className={`${blockClass}__heading-welcome`}>
                {welcomeText},{' '}
              </span>
              <span className={`${blockClass}__heading-name`}>{userName}</span>
            </h1>
          </Tooltip>
        </Column>

        {(description || tasksControllerConfig) && (
          <Column
            sm={4}
            md={8}
            lg={4}
            className={`${blockClass}__left-area-container${
              !open ? ` ${descriptionCollapsed}` : ''
            }`}>
            {description && (
              <h2 className={`${blockClass}__description`}>{description}</h2>
            )}
            <TasksController
              tasksControllerConfig={tasksControllerConfig}
              isLoading={isLoading}
              allTileGroups={allTileGroups}
              selectedTileGroup={selectedTileGroup}
              setSelectedTileGroup={setSelectedTileGroup}
            />
          </Column>
        )}

        {selectedTileGroup && (
          <Column sm={4} md={8} lg={12} className={`${blockClass}__content`}>
            {workspaceSelectorConfig?.allWorkspaces?.length && (
              <div
                className={`${blockClass}__workspace--container${
                  !open ? ` ${contentCollapsed}` : ''
                }`}>
                <WorkspaceSelector
                  workspaceSelectorConfig={workspaceSelectorConfig}
                  userName={userName}
                  isLoading={isLoading}
                />
              </div>
            )}
            <div className={`${blockClass}__tiles-container`}>
              {selectedTileGroup.tiles.map((tile) => {
                return (
                  <BaseTile
                    onClick={() => tileClickHandler?.(tile)}
                    key={tile.id}
                    id={tile.id}
                    open={open}
                    href={tile.href}
                    mainIcon={tile.mainIcon}
                    secondaryIcon={tile.secondaryIcon}
                    title={tile.title}
                    subtitle={tile.subtitle}
                    productName={productName}
                    customContent={tile.customContent}
                    isLoading={isLoading || tile.isLoading}
                    isDisabled={tile.isDisabled}
                    disabledTaskLabel={disabledTaskLabel}
                    isInteractive={tile.isInteractive}
                  />
                );
              })}
            </div>
          </Column>
        )}

        <div className={`${blockClass}__button-collapse--gradient`} />

        <div className={`${blockClass}__button-collapse--container`}>
          <Button
            id={`${blockClass}__button-collapse`}
            kind="ghost"
            renderIcon={open ? ChevronUp : ChevronDown}
            onClick={handleButtonCollapseClick}>
            {open ? collapseButtonLabel : expandButtonLabel}
          </Button>
        </div>
      </Grid>
    </section>
  );
};

(AnimatedHeader as React.FC).displayName = 'Animated Header';
(AnimatedHeader as React.FC).propTypes = {
  /**
   * Array of each tile group setup
   */
  allTileGroups: PropTypes.arrayOf(PropTypes.object),

  /**
   * Specify an optional className to be added to your Animated Header
   */
  className: PropTypes.string,

  /**
   * Custom collapse button label
   */
  collapseButtonLabel: PropTypes.string,

  /**
   * Provide short sentence in max. 3 lines related to product context
   */
  description: PropTypes.string,

  /**
   * Custom expand button label
   */
  expandButtonLabel: PropTypes.string,

  /**
   * In-product imagery / lottie animation (.json) dim. 1312 x 738
   * (to update headerAnimation content storybook requires remount in toolbar)
   */
  headerAnimation: PropTypes.object,

  /**
   * In-product imagery / static imagery dim. 1312 x 738
   * Only active when headerAnimation is not in use
   */
  headerStatic: PropTypes.object,

  /**
   * Check if is loading
   */
  isLoading: PropTypes.bool,

  /**
   * Provide current product name
   */
  productName: PropTypes.string,

  /**
   * The tile group that is active in the header
   * ex. "AI Chat Tile w/ two glass tiles", "Four glass tiles", ect.
   */
  selectedTileGroup: PropTypes.object,

  /**
   * Provide function to be called when switching selected tile group
   */
  setSelectedTileGroup: PropTypes.func,

  /**
   * Configuration for Carbon button or dropdown menu in header. Customized
   * tasks are used to allow users that have multiple roles and permissions
   * to experience better tailored content based on their need.
   * It also allows to override Carbon Button props by specifying them in tasksConfig.button.propsOverrides
   * or to override Carbon Dropdown props by specifying them in tasksConfig.dropdown.propsOverrides.
   */
  tasksControllerConfig: PropTypes.shape({
    type: PropTypes.oneOf(['button', 'dropdown']).isRequired,
    button: PropTypes.shape({
      text: PropTypes.string.isRequired,
      // Override Carbon Button props
      propsOverrides: PropTypes.object,
    }),
    dropdown: PropTypes.shape({
      allTileGroups: PropTypes.arrayOf(PropTypes.object),
      selectedTileGroup: PropTypes.object,
      setSelectedTileGroup: PropTypes.func.isRequired,
      // Override Carbon Dropdown props
      propsOverrides: PropTypes.object,
    }),
  }),

  /**
   * Handler for tile clicks
   */
  tileClickHandler: PropTypes.func,

  /**
   * Specify the current username of active user
   */
  userName: PropTypes.string,

  /**
   * Specify the current welcome text on the header
   */
  welcomeText: PropTypes.string,

  /**
   * Configuration for Carbon dropdown for workspace selection. To override Carbon Dropdown props use workspaceSelectorConfig.propsOverride
   */
  workspaceSelectorConfig: PropTypes.shape({
    // Override Carbon Dropdown props
    propsOverrides: PropTypes.object,
    allWorkspaces: PropTypes.arrayOf(PropTypes.object),
    selectedWorkspace: PropTypes.object,
    setSelectedWorkspace: PropTypes.func.isRequired,
  }),
};

export default AnimatedHeader;
