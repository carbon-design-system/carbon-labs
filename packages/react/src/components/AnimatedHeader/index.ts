/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import AnimatedHeader, {
  AriaLabels,
  TileGroup,
} from './components/AnimatedHeader/AnimatedHeader';
import HeaderTitle from './components/HeaderTitle/HeaderTitle';
import { BaseTile } from './components/Tiles/index.js';
export * from './assets';
export type {
  Workspace,
  WorkspaceSelectorConfig,
} from './components/WorkspaceSelector/WorkspaceSelector';
export type { TasksControllerConfig } from './components/TasksController/TasksController';

export {
  AnimatedHeader,
  BaseTile,
  HeaderTitle,
  type AriaLabels,
  type TileGroup,
};
