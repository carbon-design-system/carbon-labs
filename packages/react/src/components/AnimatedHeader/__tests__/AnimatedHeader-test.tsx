/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { render } from '@testing-library/react';
import 'jest-canvas-mock';
import '@testing-library/jest-dom';

import AnimatedHeader from '../components/AnimatedHeader/AnimatedHeader';
import {
  headerTiles,
  tasksControllerConfigDropdown,
  workspaceSelectorConfig,
} from '../__stories__/data';
import { TasksControllerConfig } from '../components/TasksController/TasksController';

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

jest.mock('./animated-header.scss', () => ({}));
describe('AnimatedHeader', () => {
  describe('renders as expected - Component API', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <AnimatedHeader
          tasksControllerConfig={
            tasksControllerConfigDropdown as TasksControllerConfig
          }
          workspaceSelectorConfig={workspaceSelectorConfig}
          description="Connect, monitor, and manage data."
          productName="[Product name]"
          userName="Drew"
          welcomeText="Welcome"
          selectedTileGroup={headerTiles[0]}
          setSelectedTileGroup={() => {}}
          allTileGroups={headerTiles}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
