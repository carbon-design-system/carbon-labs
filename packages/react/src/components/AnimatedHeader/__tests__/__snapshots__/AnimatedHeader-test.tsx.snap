// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedHeader renders as expected - Component API should match snapshot 1`] = `
<div>
  <section
    class="clabs--animated-header"
  >
    <div
      class="cds--css-grid"
    >
      <div
        class="clabs--animated-header__gradient--overlay"
      />
      <div
        class="clabs--animated-header__container--gradient"
      />
      <div
        class="clabs--animated-header__lottie-animation--container"
      >
        <div
          class="clabs--animated-header__lottie-animation"
        />
      </div>
      <div
        class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--css-grid-column"
      >
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <h1
              aria-labelledby="tooltip-:r0:"
              class="clabs--animated-header__heading "
            >
              <span
                class="clabs--animated-header__heading-welcome"
              >
                Welcome
                ,
                 
              </span>
              <span
                class="clabs--animated-header__heading-name"
              >
                Drew
              </span>
            </h1>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r0:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Welcome, Drew
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </div>
      <div
        class="clabs--animated-header__left-area-container cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-4 cds--css-grid-column"
      >
        <h2
          class="clabs--animated-header__description"
        >
          Connect, monitor, and manage data.
        </h2>
        <div
          class="clabs--animated-header__header-dropdown--container"
        >
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper clabs--animated-header__header-dropdown cds--dropdown__wrapper--inline cds--list-box__wrapper--inline"
          >
            <label
              class="cds--label cds--visually-hidden"
              for="downshift-:r4:-toggle-button"
              id="downshift-:r4:-label"
            >
              Label
            </label>
            <div
              class="cds--dropdown cds--dropdown--inline cds--dropdown--md cds--list-box cds--list-box--md"
              id="clabs--animated-header__header-dropdown"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r4:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r4:-label"
                class="cds--list-box__field"
                id="downshift-:r4:-toggle-button"
                role="combobox"
                tabindex="0"
                title="Customize your journey"
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  Customize your journey
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r4:-label"
                class="cds--list-box__menu"
                id="downshift-:r4:-menu"
                role="listbox"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="clabs--animated-header__content cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-12 cds--css-grid-column"
      >
        <div
          class="clabs--animated-header__workspace--container"
        >
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper clabs--animated-header__workspace cds--dropdown__wrapper--inline cds--list-box__wrapper--inline"
          >
            <label
              class="cds--label cds--visually-hidden"
              for="downshift-:r7:-toggle-button"
              id="downshift-:r7:-label"
            >
              Label
            </label>
            <div
              class="cds--dropdown cds--dropdown--inline cds--dropdown--sm cds--list-box cds--list-box--sm"
              id="clabs--animated-header__workspace"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r7:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r7:-label"
                class="cds--list-box__field"
                id="downshift-:r7:-toggle-button"
                role="combobox"
                tabindex="0"
                title=""
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  Open in: Drew's workspace
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r7:-label"
                class="cds--list-box__menu"
                id="downshift-:r7:-menu"
                role="listbox"
              />
            </div>
          </div>
        </div>
        <div
          class="clabs--animated-header__tiles-container"
        >
          <div
            class="clabs--animated-header__tile clabs--animated-header__ai-prompt-tile"
            id="clabs--animated-header__ai-prompt-tile"
          >
            <div
              class="clabs--animated-header__ai-prompt-tile--body"
            >
              <div
                class="clabs--animated-header__ai-prompt-tile--body-background"
              />
              <div
                class="clabs--animated-header__ai-prompt-tile--body-gradient"
              />
              <div
                class="clabs--animated-header__ai-prompt-tile--icons"
              >
                <svg
                  aria-hidden="true"
                  fill="var(--cds-icon-secondary)"
                  focusable="false"
                  height="24"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28,26H25V24h3V8H25V6h3a2.0023,2.0023,0,0,1,2,2V24A2.0027,2.0027,0,0,1,28,26Z"
                  />
                  <circle
                    cx="23"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="9"
                    cy="16"
                    r="2"
                  />
                  <path
                    d="M7,26H4a2.0023,2.0023,0,0,1-2-2V8A2.002,2.002,0,0,1,4,6H7V8H4V24H7Z"
                  />
                </svg>
                <div
                  class="cds--ai-label"
                  id="AILabel-:r8:"
                >
                  <span
                    class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--auto-align cds--autoalign cds--popover--bottom cds--toggletip cds--autoalign"
                  >
                    <button
                      aria-controls="id-:r9:"
                      aria-expanded="false"
                      aria-label="AI Show information"
                      class="cds--toggletip-button cds--ai-label__button cds--ai-label__button--mini cds--ai-label__button--default"
                      type="button"
                    >
                      <span
                        class="cds--ai-label__text"
                      >
                        AI
                      </span>
                    </button>
                  </span>
                </div>
              </div>
              <div
                class="clabs--animated-header__ai-prompt-tile--title"
              >
                Short description of the type of prompt
              </div>
              <div
                class="clabs--animated-header__ai-prompt-tile--text-input-container "
              >
                <div
                  class="cds--form-item clabs--animated-header__ai-prompt-tile--text-input cds--text-input-wrapper"
                >
                  <div
                    class="cds--text-input__label-wrapper"
                  >
                    <label
                      class="cds--label cds--visually-hidden"
                      dir="auto"
                      for="clabs--animated-header__ai-prompt-tile--text-input"
                    >
                      AI Chat Input
                    </label>
                  </div>
                  <div
                    class="cds--text-input__field-outer-wrapper"
                  >
                    <div
                      class="cds--text-input__field-wrapper"
                    >
                      <input
                        class="cds--text-input cds--text-input--sm cds--layout--size-sm"
                        id="clabs--animated-header__ai-prompt-tile--text-input"
                        placeholder="Start chatting..."
                        title="Start chatting..."
                        type="text"
                        value=""
                      />
                      <span
                        aria-atomic="true"
                        aria-live="assertive"
                        class="cds--text-input__counter-alert"
                        role="alert"
                      />
                    </div>
                  </div>
                </div>
                <span
                  class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top-end cds--tooltip cds--icon-tooltip cds--icon-tooltip--disabled"
                >
                  <div
                    class="cds--tooltip-trigger__wrapper"
                  >
                    <button
                      aria-labelledby="tooltip-:rc:"
                      class="cds--btn--icon-only clabs--animated-header__ai-prompt-tile--icon-button  cds--btn cds--btn--sm cds--layout--size-sm cds--btn--ghost cds--btn--disabled"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.45,15.11l-22-11a1,1,0,0,0-1.08.12,1,1,0,0,0-.33,1L7,16,4,26.74A1,1,0,0,0,5,28a1,1,0,0,0,.45-.11l22-11a1,1,0,0,0,0-1.78Zm-20.9,10L8.76,17H18V15H8.76L6.55,6.89,24.76,16Z"
                        />
                      </svg>
                    </button>
                  </div>
                  <span
                    aria-hidden="true"
                    class="cds--popover"
                    id="tooltip-:rc:"
                    role="tooltip"
                  >
                    <span
                      class="cds--popover-content cds--tooltip-content"
                    >
                      Chat in [Product name]
                    </span>
                    <span
                      class="cds--popover-caret"
                    />
                  </span>
                </span>
              </div>
              <div
                class="clabs--animated-header__ai-prompt-tile--footer"
              >
                <a
                  class="cds--btn cds--btn--sm cds--layout--size-sm cds--btn--ghost"
                  href="#"
                >
                  Open 
                  [Product name]
                </a>
              </div>
            </div>
          </div>
          <a
            class="cds--link clabs--animated-header__tile clabs--animated-header__glass-tile"
            href="#"
          >
            <div
              class="clabs--animated-header__glass-tile--body"
            >
              <div
                class="clabs--animated-header__glass-tile--body-background"
              />
              <div
                class="clabs--animated-header__glass-tile--icons"
              >
                <svg
                  aria-hidden="true"
                  fill="var(--cds-icon-secondary)"
                  focusable="false"
                  height="24"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25 12L25 3 23 3 23 5 20 5 20 7 23 7 23 12 20 12 20 14 28 14 28 12 25 12zM8.5 5c1.9299 0 3.5 1.5701 3.5 3.5s-1.5701 3.5-3.5 3.5-3.5-1.5701-3.5-3.5 1.5701-3.5 3.5-3.5m0-2c-3.0376 0-5.5 2.4624-5.5 5.5s2.4624 5.5 5.5 5.5 5.5-2.4624 5.5-5.5-2.4624-5.5-5.5-5.5h0zM8.5 20c1.9299 0 3.5 1.5701 3.5 3.5s-1.5701 3.5-3.5 3.5-3.5-1.5701-3.5-3.5 1.5701-3.5 3.5-3.5m0-2c-3.0376 0-5.5 2.4624-5.5 5.5s2.4624 5.5 5.5 5.5 5.5-2.4624 5.5-5.5-2.4624-5.5-5.5-5.5h0zM23.5 20c1.9299 0 3.5 1.5701 3.5 3.5s-1.5701 3.5-3.5 3.5-3.5-1.5701-3.5-3.5 1.5701-3.5 3.5-3.5m0-2c-3.0376 0-5.5 2.4624-5.5 5.5s2.4624 5.5 5.5 5.5 5.5-2.4624 5.5-5.5-2.4624-5.5-5.5-5.5h0z"
                  />
                </svg>
              </div>
              <div
                class="clabs--animated-header__glass-tile--title"
              >
                Load data
              </div>
              <div
                class="clabs--animated-header__glass-tile--footer"
              >
                <div
                  class="clabs--animated-header__glass-tile--subtitle"
                >
                  with Data explorer
                </div>
              </div>
            </div>
          </a>
          <a
            class="cds--link clabs--animated-header__tile clabs--animated-header__glass-tile"
            href="#"
          >
            <div
              class="clabs--animated-header__glass-tile--body"
            >
              <div
                class="clabs--animated-header__glass-tile--body-background"
              />
              <div
                class="clabs--animated-header__glass-tile--icons"
              >
                <svg
                  aria-hidden="true"
                  fill="var(--cds-icon-secondary)"
                  focusable="false"
                  height="24"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.06,17.88A4.25,4.25,0,0,0,11,18a4,4,0,0,0,2.23-.68l3.22,2.87a3.88,3.88,0,0,0-.2,3.17A4,4,0,1,0,22.62,19l2.54-5.09a3.78,3.78,0,0,0,2.91-.53A4,4,0,1,0,23.38,13l-2.54,5.09A3.86,3.86,0,0,0,20,18a4,4,0,0,0-2.23.68l-3.22-2.87a3.88,3.88,0,0,0,.2-3.17A4,4,0,1,0,8.3,16.93L4,25V2H2V28a2,2,0,0,0,2,2H30V28H4.67ZM26,8a2,2,0,1,1-2,2A2,2,0,0,1,26,8ZM22,22a2,2,0,1,1-2-2A2,2,0,0,1,22,22ZM11,12a2,2,0,1,1-2,2A2,2,0,0,1,11,12Z"
                  />
                </svg>
              </div>
              <div
                class="clabs--animated-header__glass-tile--title"
              >
                Monitor data performance and system capacity
              </div>
              <div
                class="clabs--animated-header__glass-tile--footer"
              >
                <div
                  class="clabs--animated-header__glass-tile--subtitle"
                >
                  with Monitoring hub
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="clabs--animated-header__button-collapse--gradient"
      />
      <div
        class="clabs--animated-header__button-collapse--container"
      >
        <button
          class="cds--btn cds--btn--ghost"
          id="clabs--animated-header__button-collapse"
          type="button"
        >
          Collapse
          <svg
            aria-hidden="true"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
            />
          </svg>
        </button>
      </div>
    </div>
  </section>
</div>
`;
